From 56fa5710ad669fe673ff9d1c0f609226a6cd55ad Mon Sep 17 00:00:00 2001
Message-Id: <56fa5710ad669fe673ff9d1c0f609226a6cd55ad.1413837857.git.chang-joon.lee@intel.com>
In-Reply-To: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
References: <006497b7323eca32c2f489a1a9f662743c0d16d2.1413837855.git.chang-joon.lee@intel.com>
From: Daniel Vetter <daniel.vetter@ffwll.ch>
Date: Wed, 25 Jun 2014 22:02:04 +0300
Subject: [PATCH 147/312] FROM_UPSTREAM [VPG]: drm/i915: ddi: enable runtime
 pm during dpms

FOR: LCK-1074
Change-Id: I0edca979863cc49d3c86b04bf6fe06e15e0d75be
Signed-off-by: Imre Deak <imre.deak@intel.com>
Reviewed-by: Paulo Zanoni <paulo.r.zanoni@intel.com>
[danvet: Remove now bogus comment.]
Signed-off-by: Daniel Vetter <daniel.vetter@ffwll.ch>
---
 drivers/gpu/drm/i915/intel_display.c |   28 ++++++++--------------------
 1 file changed, 8 insertions(+), 20 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 9bf6ad0..e5f5ff7 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -5495,32 +5495,20 @@ void intel_crtc_update_dpms(struct drm_crtc *crtc)
 
 	if (enable) {
 		if (!intel_crtc->active) {
-			/*
-			 * FIXME: DDI plls and relevant code isn't converted
-			 * yet, so do runtime PM for DPMS only for all other
-			 * platforms for now.
-			 */
-			if (!HAS_DDI(dev)) {
-				domains = get_crtc_power_domains(crtc);
-				for_each_power_domain(domain, domains)
-					intel_display_power_get(dev_priv,
-								domain);
-				intel_crtc->enabled_power_domains = domains;
-			}
-
+			domains = get_crtc_power_domains(crtc);
+			for_each_power_domain(domain, domains)
+				intel_display_power_get(dev_priv, domain);
+			intel_crtc->enabled_power_domains = domains;
 			dev_priv->display.crtc_enable(crtc);
 		}
 	} else {
 		if (intel_crtc->active) {
 			dev_priv->display.crtc_disable(crtc);
 
-			if (!HAS_DDI(dev)) {
-				domains = intel_crtc->enabled_power_domains;
-				for_each_power_domain(domain, domains)
-					intel_display_power_put(dev_priv,
-								domain);
-				intel_crtc->enabled_power_domains = 0;
-			}
+			domains = intel_crtc->enabled_power_domains;
+			for_each_power_domain(domain, domains)
+				intel_display_power_put(dev_priv, domain);
+			intel_crtc->enabled_power_domains = 0;
 		}
 	}
 
-- 
1.7.9.5

